### ISSUE TITLE
Different SplashScreen for Light and Dark Modes

### ISSUE THREAD

**Issue by jrahma** [dotnet/maui#9578]
### Description

Hi,

It will be great if we will be able to set a different ResourceDictionary for the Splash Screen when light and Dark..

### Public API Changes

```
#if AppTheme is Light
<MauiSplashScreen Include="Resources\Splash\splash_light.svg" Color="#FFFFFF" BaseSize="128,128" />

#if AppTheme is Dark
<MauiSplashScreen Include="Resources\Splash\splash_dark.svg" Color="#000000" BaseSize="128,128" />

```

### Intended Use-Case

To set the AppTheme's Splash Screen

**Comment by Cybrosys** [comment]
I would also like to request the ability to specify different splash screens depending on the theme itself. The reason behind this is that we do different brandings for the same app depending on the end customer. 

We currently read a setting value indicating which theme that should be used, and on iOS change the image path in the storyboard and on Android change the splash activity theme used.

**Comment by mattleibow** [comment]
Yeah, this is pretty important - and also for icons and general drawables on the platforms.

**Comment by danielftz** [comment]
I would also like to see this. The splashscreen on Android especailly still needs some work

**Comment by SarthakB26** [comment]
Hi I am also seeing that SplashScreen Background color changes to Black when app open in dark mode no matter what I set in csproj

**<MauiSplashScreen Include="Resources\Splash\splashscreen.svg" Color="#512BD4" />**

here Color="#512BD4"  has no effect. Color auto changes to black background when app open in darkmode. Works fine in lightmode

Is there any way set keep same color in darkmode as well that we set in csproj file for SplashScreen
Please help

**Comment by mattleibow** [comment]
@SarthakB26 are you able to open a new issue with the background color not working? Some info that may help us diagnose your issue is a repro with the image you are using.

Also, the android device and os version and maui version. 

I just ran a maui app now in dark mode on my Samsung and I got the default putple splash with white logo.

**Comment by SarthakB26** [comment]
> 

sure will do and inform here

**Comment by harry-cpp** [comment]
Workaround for Android if anyone needs it.

Set the color parameter of the splash image in .csproj to Transparent:
```xml
<MauiSplashScreen Include="Resources\Splash\splash.svg" Color="#00000000" BaseSize="128,128" />
```

The Android theme MAUI creates is using `@color/maui_splash_color` for its background color: 
https://github.com/dotnet/maui/blob/c92d44c68fe81c57ef8caec2506e6788309b8ff4/src/Core/src/Platform/Android/Resources/values/styles.xml#L37 this is tweakable from the app side.

Add the following in Platforms/Android/Resources/values/colors.xml:
```xml
<color name="maui_splash_color">#ffffff</color>
```

For dark mode, Android will automatically use `values-night` if they are available, so all you've got to do now is copy over the file above to a `values-night` folder, that is: `Platforms/Android/Resources/values-night/colors.xml`, and tweak the colors in there for your desired night color choices.

**Comment by SarthakB26** [comment]
Thanks for the workaround @harry-cpp . I also found this issue comes when using #ffffff specifically as a splash screen background color wasn't happening on empty maui app which is using some blue color rather then white.
As a workaround I used a different shade of white color.

**Comment by Hugo64** [comment]
@harry-cpp This solution is wonderful! Thanks! Congratulations! I'm looking for iOS right now.

**Comment by aabc123** [comment]
Has anyone found a workaround for iOS also?  We're in the process of porting our Xamarin app to Maui but need this as the splash screen background colour is also the colour of the main app page which is different for dark/light mode.

**Comment by PhilippRoessner** [comment]
If anyone is interested, I found a solution to use the image in the splashscreen depending on the current configuration. We are whitelabeling our app so each customer has it's on branding. I added the following to the maui.csproj file right before the </Project> Tag:

```
	<Target Name="CopySplashFile" BeforeTargets="Build">
		<ItemGroup>
			<!-- Use the dynamically determined environment name in the path -->
			<SplashFiles Include="$(SolutionDir)\CustomerPortal.WebShared\wwwroot\environmentAssets\$(Configuration)\logo.png" />
		</ItemGroup>
		<Copy SourceFiles="@(SplashFiles)" DestinationFiles="$(MSBuildProjectDirectory)\Resources\Splash\splash.png" />
	</Target>
```

CustomerPortal.WebShared is a Razor Class Library Project (this is a Maui Hybrid Project).

We create a Configuration Profile (Release,Debug,Customer1,Customer2) for each customer and depending on the name, we are loading defferent settings files and assets

**Comment by LeoJHarris** [comment]
> I would also like to request the ability to specify different splash screens depending on the theme itself. The reason behind this is that we do different brandings for the same app depending on the end customer.
> 
> We currently read a setting value indicating which theme that should be used, and on iOS change the image path in the storyboard and on Android change the splash activity theme used.

Same here on different build configurations, there really needs to be some proper guidance/documentation on how to set this up, not all app will have a single splash screen to load up.

**Comment by Moes81** [comment]
So...any new iformation on how to create light/dark splashscreens for iOS that contain actual images specific for light/dark mode? I can create a storyboard that supports light/dark but since asset catalogs are not supported by maui, the images are not loaded. And, there is no documentation whatsoever about how to declare images specifically for light/dark on iOS only.

Any help is appreciated.

**Comment by mattleibow** [comment]
@rolfbjarne the asset catalogs should be working with maui? @Moes81 did you put the assets under the Platforms/iOS folder?

**Comment by rolfbjarne** [comment]
> [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/249268?s=64&amp;v=4">@rolfbjarne](https://github.com/rolfbjarne) the asset catalogs should be working with maui?

asset catalogs should work just fine

**Comment by Axemasta** [comment]
For anyone interested I have a [sample repo](https://github.com/Axemasta/SuperSplash) showing you how to setup the native splashscreen to use assets (colours) and also images from your maui project within the LaunchScreen.storyboard file.

Anything in the `Platforms/iOS/Resources` folder automatically gets pulled through as a `BundleResource` so its just a case of building the file in xcode, creating an assets catalog for whatever you need and drag + dropping into your maui project.
